ext/ffi/lib.rs

```diff
@@ -1315,6 +1436,19 @@ fn ffi_call(
           pointer: cif.call::<*const u8>(fun_ptr, &call_args),
         }
       }
+      NativeType::Struct(_) => {
+        let size = result_type.get_size();
+        let mut result = vec![0u8; size].into_boxed_slice();
+        libffi::raw::ffi_call(
+          cif.as_raw_ptr(),
+          Some(*fun_ptr.as_safe_fun()),
+          result.as_mut_ptr() as *mut c_void,
+          call_args.as_ptr() as *mut *mut c_void,
+        );
+        NativeValue {
+          struct_value: Box::into_raw(result),
+        }
+      }
```

Comment:
This is similar to the code on lines 1355; can this be hoisted out into its own function?